ballista/scheduler/src/scheduler_server/grpc.rs

```diff
@@ -70,7 +72,10 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
                 "Bad request because poll work is not supported for push-based task scheduling",
             ));
         }
-        let remote_addr = request.remote_addr();
+        let remote_addr = request
+            .extensions()
+            .get::<ConnectInfo<SocketAddr>>()
+            .cloned();
```

Comment:
nit: this code block is repeated a few times so it may be worth extracting into a method